public/app/features/profile/ChangePasswordForm.tsx

```diff
@@ -1,75 +1,70 @@
-    this.setState({ oldPassword });
-  };
-
-  onNewPasswordChange = (newPassword: string) => {
-    this.setState({ newPassword });
-  };
-
-  onConfirmPasswordChange = (confirmNew: string) => {
-    this.setState({ confirmNew });
-  };
+export const ChangePasswordForm: FC<Props> = ({ onChangePassword, isSaving }) => {
+  const { ldapEnabled, authProxyEnabled } = config;
 
-  onSubmitChangePassword = (event: MouseEvent<HTMLInputElement>) => {
-    event.preventDefault();
-    this.props.onChangePassword({ ...this.state });
-  };
-
-  render() {
-    const { oldPassword, newPassword, confirmNew } = this.state;
-    const { isSaving } = this.props;
-    const { ldapEnabled, authProxyEnabled } = config;
-
-    if (ldapEnabled || authProxyEnabled) {
-      return <p>You cannot change password when ldap or auth proxy authentication is enabled.</p>;
-    }
-
-    return (
-      <form name="userForm" className="gf-form-group">
-        <div className="gf-form max-width-30">
-          <PasswordInput label="Old Password" onChange={this.onOldPasswordChange} value={oldPassword} />
-        </div>
-        <div className="gf-form max-width-30">
-          <PasswordInput label="New Password" onChange={this.onNewPasswordChange} value={newPassword} />
-        </div>
-        <div className="gf-form max-width-30">
-          <PasswordInput label="Confirm Password" onChange={this.onConfirmPasswordChange} value={confirmNew} />
-        </div>
-        <div className="gf-form-button-row">
-          <Button variant="primary" onClick={this.onSubmitChangePassword} disabled={isSaving}>
-            Change Password
-          </Button>
-          <LinkButton variant="secondary" href={`${config.appSubUrl}/profile`}>
-            Cancel
-          </LinkButton>
-        </div>
-      </form>
-    );
+  if (ldapEnabled || authProxyEnabled) {
+    return <p>You cannot change password when ldap or auth proxy authentication is enabled.</p>;
   }
-}
+  return (
+    <div
+      className={css`
+        max-width: 400px;
+      `}
+    >
+      <Form onSubmit={onChangePassword}>
+        {({ register, errors, getValues }) => {
+          return (
+            <>
+              <Field label="Old password" invalid={!!errors.oldPassword} error={errors?.oldPassword?.message}>
+                <Input type="password" name="oldPassword" ref={register({ required: 'Old password is required' })} />
+              </Field>
+
+              <Field label="New password" invalid={!!errors.newPassword} error={errors?.newPassword?.message}>
+                <Input
+                  type="password"
+                  name="newPassword"
+                  ref={register({
+                    required: 'New password is required',
+                    validate: {
+                      confirm: v => v === getValues().confirmNew || 'Passwords must match',
+                      old: v => v !== getValues().oldPassword || `New password can't be the same as the old one.`,
```

Comment:
Is this some new validation? Makes sense, just asking

Reply:
No, it's a part of `react-hook-form`, don't know since when though. Really nice feature!